clang/include/clang/Driver/Options.td

```diff
@@ -3785,7 +3785,14 @@ def ftarget_export_symbols : Flag<["-"], "ftarget-export-symbols">,
 def fno_target_export_symbols : Flag<["-"], "fno-target-export-symbols">,
   Visibility<[ClangOption, CLOption, DXCOption]>;
 def : Flag<["-"], "fsycl-rdc">, Visibility<[ClangOption, CLOption, DXCOption]>, Alias<fgpu_rdc>;
-def : Flag<["-"], "fno-sycl-rdc">, Visibility<[ClangOption, CLOption, DXCOption]>, Alias<fno_gpu_rdc>;
+def : Flag<["-"], "fno-sycl-rdc">,
+  Visibility<[ClangOption, CLOption, DXCOption]>, Alias<fno_gpu_rdc>,
+  HelpText<"Generate relocatable device code during SYCL offload target "
```

Comment:
Do we need an error check if someone tries to pass `-fno-sycl-rdc` and `-c` but is not compiling for `spir64_gen`? Or is it we only do the early linking if it's AOT but allow the previous flow if it's JIT?

Reply:
The early AOT linking only occurs for AOT targets.  JIT is not impacted.